perceval/backends/mozilla_reps.py

```diff
@@ -0,0 +1,263 @@
+                if not events_next_uri:
+                    more_events = False
+                events = events_data['objects']
+            except:
+                cause = ("Bad JSON format for mozilla_events: %s" % (events_data))
+                raise ParseError(cause=cause)
+            for rep in events:
+                yield rep
+                nevents += 1
+            self._flush_cache_queue()
+
+        logger.info("Total number of events: %i (%i expected)", nevents, nevents_total)
+
+    @metadata
+    def fetch_from_cache(self):
+        """Fetch the builds from the cache.
+
+        :returns: a generator of events
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = self.cache.retrieve()
+
+        for items in cache_items:
+            events = json.loads(items)['objects']
+            for event in events:
+                yield event
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Build item."""
+        return MozillaReps.__metadata_id_event(item)
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a MozillaReps item.
+
+        The timestamp is extracted from 'timestamp' field.
+        This date is a UNIX timestamp but needs to be converted to
+        a float value.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        return MozillaReps.__metadata_updated_on_event(item)
+
+    @staticmethod
+    def __metadata_id_event(item):
+        return str(item['event_url'])
+
+    @staticmethod
+    def __metadata_updated_on_event(item):
+        return float(parser.parse(item['end']).timestamp())
+
+
+class MozillaRepsClient:
+    """MozillaReps API client.
+
+    This class implements a simple client to retrieve builds from
+    projects in a MozillaReps node.
+
+    :param url: URL of MozillaReps url https://reps.mozilla.org
+
+    :raises HTTPError: when an error occurs doing the request
+    """
+
+    MAX_ITEMS_EV = "40"  # max number of events per iteration
+
+    def __init__(self, url):
+        self.url = url
+        self.api_events_url = urljoin(self.url, '/api/v1/event/')
+        self.api_events_url += '/'  # API needs a final /
+
+    def call(self, params):
+        """Run an API command.
+        :param cgi: cgi command to run on the server
+        :param params: dict with the HTTP parameters needed to run
+            the given command
+        """
+        logger.debug("MozillaReps client calls API: %s params: %s",
+                     self.api_events_url, str(params))
+
+        req = requests.get(self.api_events_url, params=params)
+        req.raise_for_status()
+
+        return req.text
+
+    def get_events(self,  next_url=None):
```

Comment:
Yes, it is used but in the last changes, I have not reflect the next_url case in the param logic!

Fixed!